defold-spine/src/comp_spine_model.cpp

```diff
@@ -800,6 +810,41 @@ namespace dmSpine
         dmRender::AddToRender(render_context, &ro);
     }
 
+    static void RenderListFrustumCulling(dmRender::RenderListVisibilityParams const &params)
+    {
+        DM_PROFILE("SpineModel");
+
+        SpineModelWorld* spine_world = (SpineModelWorld*)params.m_UserData;
+
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            int component_index = entry->m_UserData;
+
+            SpineModelComponent* component_p = spine_world->m_Components.Get(component_index);
```

Comment:
Ah, no.
The index you put in the user data is the internal vector index.

The index used by the Get() function, needs the logical index. I.e. the one returned from `dmObjectPool::Alloc()`.

I suggest going back to the previous way of using the internal index to directly index the internal array. I.e. the same way you iterate on the array when creating the render items.

That was the reason why we moved the code from Update() to Render(), so that you could be certain that the internal index would still be valid.

Reply:
Fixed it. Also, removed a forgotten "volatile" specifier in a variable. I had used this to be able to use watches in gdb.